pkg/controller/deployment/deployment_controller.go

```diff
@@ -749,3 +913,15 @@ func (dc *DeploymentController) updateDeployment(deployment *extensions.Deployme
 	// TODO: Using client for now, update to use store when it is ready.
 	return dc.expClient.Deployments(deployment.ObjectMeta.Namespace).Update(deployment)
 }
+
+func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deployment, rc *api.ReplicationController) (*extensions.Deployment, error) {
+	dTemplateHash := podutil.GetPodTemplateSpecHash(deployment.Spec.Template)
+	rcTemplateHash := podutil.GetPodTemplateSpecHash(*rc.Spec.Template)
+	if dTemplateHash != rcTemplateHash {
+		glog.Infof("Rolling back deployment %s to template spec %+v", deployment.Name, rc.Spec.Template.Spec)
+		deployment.Spec.Template.Spec = rc.Spec.Template.Spec
+	} else {
+		glog.V(4).Infof("Rolling back to a revision that contains the same template as current deployment %s, skipping rollback...", deployment.Name)
+	}
+	return dc.cleanUpRollbackTo(deployment)
```

Comment:
You are using `cleanUpRollbackTo` not only to clean up the rollback config but also to update the deployment template. Not sure I like this indirection, just assign rollbackTo to nil and use a plain deployment update.